src/raftstore/store/worker/read.rs

```diff
@@ -457,59 +590,13 @@ impl<C: Sender<StoreMsg>> Runnable<Task> for LocalReader<C> {
     }
 
     fn run_batch(&mut self, tasks: &mut Vec<Task>) {
-        let mut sent = None;
         self.metrics
             .borrow()
             .batch_requests_size
             .observe(tasks.len() as _);
 
-        for task in tasks.drain(..) {
-            match task {
-                Task::Register(delegate) => {
-                    debug!(
-                        "{} register ReadDelegate for {:?}",
-                        delegate.tag, delegate.peer_id
-                    );
-                    self.delegates.insert(delegate.region.get_id(), delegate);
-                }
-                Task::Read(StoreMsg::RaftCmd {
-                    send_time,
-                    request,
-                    callback,
-                }) => {
-                    self.propose_raft_command(request, callback, send_time);
-                    if sent.is_none() {
-                        sent = Some(send_time);
-                    }
-                }
-                Task::Read(StoreMsg::BatchRaftSnapCmds {
-                    send_time,
-                    batch,
-                    on_finished,
-                }) => {
-                    self.propose_batch_raft_snapshot_command(batch, on_finished);
-                    if sent.is_none() {
-                        sent = Some(send_time);
-                    }
-                }
-                Task::Read(other) => {
-                    unimplemented!("unsupported Msg {:?}", other);
-                }
-                Task::Update((region_id, progress)) => {
-                    if let Some(delegate) = self.delegates.get_mut(&region_id) {
-                        delegate.update(progress);
-                    } else {
-                        panic!(
-                            "unregistered ReadDelegate, region_id: {}, {:?}",
-                            region_id, progress
-                        );
-                    }
-                }
-                Task::Destroy(region_id) => {
-                    self.delegates.remove(&region_id);
-                }
-            }
-        }
+        let sent = self.handle_tasks(tasks);
+        self.handle_ready_tasks(tasks);
```

Comment:
Seems rename `handle_tasks` to `pre_handle_tasks`, and rename `handle_ready_tasks` to `handle_task` is better.